### PR TITLE
fix(wazero): remove panic from offchain submit transaction

### LINKED ISSUE #3853: (dot/sync): Node panics in runtime when submitting equivacation report
## Issue summary

- Sometimes when running the node in tip sync mode we get this error from the runtime
- This issue will take investigating into the root cause, however, I do not think it is actually an issue with submitting equivocation and is instead a broader problem (I think just on restart)
- This might be resolved by recent work on graceful shutdown, however, that is still to be seen so this needs to be investigated

## Other information and links

image

### PR DESCRIPTION
**Pull request by jimjbrettj**
## Changes

When calling ext_offchain_submit_transaction_version_1 our node would panic. This was due to us not initializing the TransactionState causing it to be nil. However, fixing this highlighted that we do not correctly handle this function, though this is outside this PR's scope (see links below for issue representing this work effort). 

This PR simply initialized the TransactionState (to prevent future bugs caused by this) and removes the problematic code (since it is optional) and leaves the further off-chain workers investigation to future work represented by the issue linked above.

To summarize, this PR fixes the panic happening during tip sync, and two issues have been created to represent future work.
1) Offchain worker investigation #3986
2) Equivocation investigation #3987

## Issues

Closes: #3853, #3706

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]
Nice description for this PR.  Please remove command from `Tests` in description since no tests have been added to confirm this update.

**Review by dimartiro — APPROVED** [review]
lgtm
